### PR TITLE
added 'API as a Product' principle

### PR DESCRIPTION
**Pull request by tfrauenstein**
- added 'API as a Product' as new principle
- editorial lifting of API Design Principles

### CODE REVIEW

File: design-principles/DesignPrinciples.md [hunk 1]
```diff
@@ -1,33 +1,64 @@
...
-1. We prefer REST-based APIs with JSON payloads
-2. We prefer systems to be truly RESTful
-3. We strive to build interoperating distributed systems that different teams can evolve in parallel
+Comparing SOA web service interfacing style of SOAP vs. REST,
+the former tend to be centered around operations that are usually use-case specific and specialized.
+In contrast, REST is centered around business (data) entities exposed as resources 
+that are identified via URIs and can be manipulated via standardized CRUD-like methods 
+using different representations, self-descriptive messages and hypermedia. 
+RESTful APIs tend to be less use-case specific, comes with less rigid client / server coupling 
+and are more suitable as platform interface being open for diverse client applications.
```
Comment (dehora) [inline design-principles/DesignPrinciples.md]: edit: `comes with` -> `come with`
edit: `as platform ` -> `as a platform `

File: design-principles/DesignPrinciples.md [hunk 2]
```diff
@@ -1,33 +1,64 @@
...
-  Design](http://restful-api-design.readthedocs.org/en/latest/)
+* Book: [Build APIs You Won't Hate](https://leanpub.com/build-apis-you-wont-hate)
+* InfoQ eBook: [Web APIs: From Start to Finish](http://www.infoq.com/minibooks/emag-web-api)
+* Lessons-learned blog: [Thoughts on RESTful API Design](http://restful-api-design.readthedocs.org/en/latest/)
+* Fielding Dissertation: [Architectural Styles and the Design of Network-Based Software Architectures](http://www.ics.uci.edu/~fielding/pubs/dissertation/top.htm)
+
+
+## API as a Product
+
+As mentioned above, Zalando transforms from an online shop into an expansive fashion platform comprising a
```
Comment (dehora) [inline design-principles/DesignPrinciples.md]: edit: `transforms from an` -> `has transformed from` or `is transforming from` depending on how you see progress ;)

Comment (tfrauenstein) [inline design-principles/DesignPrinciples.md]: done

File: design-principles/DesignPrinciples.md (hunk 1 above)
Comment (tfrauenstein) [inline design-principles/DesignPrinciples.md]: done

**Comment by dehora** [comment]
:+1:

File: design-principles/DesignPrinciples.md [hunk 3]
```diff
@@ -1,33 +1,63 @@
...
+
+As mentioned above, Zalando is transforming from an online shop into an expansive fashion platform 
+comprising a rich set of products following a Software as a Platform (SaaS) model for our business partners. 
+As a company we want to deliver products to our (internal and external) customers which can be consumed like a service. 
+
+Platform products provide their functionality via (public) APIs; 
+hence, the design of our APIs should be based on the API as a Product principle:
+
+* Treat your API as product and understand the needs of its customers
+* Take ownership and advocate for the customer and continual improvement
```
Comment (whiskeysierra) [inline design-principles/DesignPrinciples.md]: > and continuous improvement

https://www.vocabulary.com/articles/chooseyourwords/continual-continuous/

Comment (tfrauenstein) [inline design-principles/DesignPrinciples.md]: thx. -- done

**Comment by tfrauenstein** [comment]
👍

**Comment by tlossen** [comment]
👍

**Pull request merged**